Makefile

```
@@ -305,41 +203,46 @@ else
-install_cert_manager:
-	kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/v1.0.4/cert-manager.yaml
 
-_operator_sdk:
-	@{ \
-		if ! command -v operator-sdk &> /dev/null; then \
-			echo 'operator-sdk $(OPERATOR_SDK_VERSION) is expected to be used for this target but it is not installed' ;\
-			exit 1 ;\
-		else \
-			SDK_VER=$$(operator-sdk version | cut -d , -f 1 | cut -d : -f 2 | cut -d \" -f 2) && \
-			if [ "$${SDK_VER}" != $(OPERATOR_SDK_VERSION) ]; then \
-				echo "WARN: operator-sdk $(OPERATOR_SDK_VERSION) is expected to be used for this target but $${SDK_VER} found" \
-				echo "WARN: Please use the recommended operator-sdk if you face any issue" \
-			fi \
-		fi \
-	}
-ifneq ($(shell operator-sdk version | cut -d , -f 1 | cut -d : -f 2 | cut -d \" -f 2),$(OPERATOR_SDK_VERSION))
-	@echo 'WARN: operator-sdk $(OPERATOR_SDK_VERSION) is expected to be used for this target but $(shell operator-sdk version | cut -d , -f 1 | cut -d : -f 2 | cut -d \" -f 2) found.'
-	@echo 'WARN: Please use the recommended operator-sdk if you face any issue.'
-endif
-
-_check_cert_manager:
-ifeq ($(PLATFORM),kubernetes)
-	if ! ${K8S_CLI} api-versions | grep -q '^cert-manager.io/v1$$' ; then \
-		echo "Cert-manager is required for deploying on Kubernetes. See 'make install_cert_manager'" ;\
-		exit 1 ;\
-	fi
-endif
+### compile-devworkspace-controller: Compiles the devworkspace-controller binary
+.PHONY: compile-devworkspace-controller
+compile-devworkspace-controller:
+	export ARCH="$(shell uname -m)";  \
+	if [[ "$(ARCH)" == "x86_64" ]]; then  \
+		export ARCH="amd64";  \
+	elif [[ "$(ARCH)" == "aarch64" ]]; then  \
+		export ARCH="arm64";  \
+	fi  \
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) GO111MODULE=on go build \
+	-a -o _output/bin/devworkspace-controller \
+	-gcflags all=-trimpath=/ \
+	-asmflags all=-trimpath=/ \
+	-ldflags "-X $(GO_PACKAGE_PATH)/version.Commit=$(GIT_COMMIT_ID) \
+	-X $(GO_PACKAGE_PATH)/version.BuildTime=$(BUILD_TIME)" \
+	main.go
+
+### compile-webhook-server: Compiles the webhook-server
+.PHONY: compile-webhook-server
+compile-webhook-server:
+	export ARCH="$(shell uname -m)";  \
+	if [[ "$(ARCH)" == "x86_64" ]]; then  \
+		export ARCH="amd64";  \
+	elif [[ "$(ARCH)" == "aarch64" ]]; then  \
+		export ARCH="arm64";  \
+	fi  \
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) GO111MODULE=on go build \
+	-o _output/bin/webhook-server \
+	-gcflags all=-trimpath=/ \
+	-asmflags all=-trimpath=/ \
+	-ldflags "-X $(GO_PACKAGE_PATH)/version.Commit=$(GIT_COMMIT_ID) \
+	-X $(GO_PACKAGE_PATH)/version.BuildTime=$(BUILD_TIME)" \
+	webhook/main.go
 
 .PHONY: help
-### help: print this message
+### help: Prints this message
 help: Makefile
 	@echo 'Available rules:'
-	@sed -n 's/^### /    /p' $< | awk 'BEGIN { FS=":" } { printf "%-30s -%s\n", $$1, $$2 }'
+	@sed -n 's/^### /    /p' $< deploy.mk | awk 'BEGIN { FS=":" } { printf "%-30s -%s\n", $$1, $$2 }'
```

I like your suggestion of using `$(MAKEFILE_LIST)` here -- that way the help stays accurate if `deploy.mk` is not included.
```suggestion
	@sed -n 's/^### /    /p' $(MAKEFILE_LIST) | awk 'BEGIN { FS=":" } { printf "%-30s -%s\n", $$1, $$2 }'
```

---

Okay :-) done!